src/bgw/job_stat.c

```diff
@@ -346,7 +347,8 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 																		  consecutive_failures);
 	Assert(consecutive_failures > 0 && multiplier < 63);
 
-	MemoryContext oldctx;
+	MemoryContext oldctx = CurrentMemoryContext;
+	ResourceOwner oldowner = CurrentResourceOwner;
```

Comment:
would it be possible to place it inside the `PG_TRY` - so that its guarded by the try?

I think in some odd cases `BeginInternalSubTransaction` may error out...possibly leaving it in inconsistent state

Reply:
Hmm... maybe. Will investigate.

Reply:
Yes, it can error out and it is better to move the call to `BeginInternalSubTransaction()` inside the `PG_TRY`.

Reply:
I don't see `CurrentResourceOwner` changed anywhere - I'm just wondering how does that change?

Reply:
It is changed inside `BeginInternalSubTransaction`, `ReleaseCurrentSubTransaction`, and `RollbackAndReleaseCurrentSubTransaction`.